src/envoy/http/backend_auth/filter.cc

```diff
@@ -54,9 +54,14 @@ FilterHeadersStatus Filter::decodeHeaders(RequestHeaderMap& headers, bool) {
   // Make sure route is calculated
   auto route = decoder_callbacks_->route();
 
-  // This shouldn't happen as the fallback wildcard route match should catch all
+  // This shouldn't happen as the catch-all route match should catch all
   // the undefined requests.
-  ASSERT(route != nullptr && route->routeEntry() != nullptr);
+  ASSERT(route != nullptr);
```

Comment:
Actually,  I don't know when route() return null and when route()->routeEntry() return null.   I am not sure if it is safe to assert route != null, but not to assert route->routeEntry().

This behaviors may change from upstream envoy.

To be safe:  if either is null, just return continue.


Let us

Reply:
You are right. For `routeEntry` is null for catch-all as we don't put any per-route-config there. I think it is find to `continue`.